source/common/http/conn_pool_grid.cc

```diff
@@ -349,11 +349,44 @@ bool ConnectivityGrid::isPoolHttp3(const ConnectionPool::Instance& pool) {
   return &pool == pools_.begin()->get();
 }
 
-bool ConnectivityGrid::isHttp3Broken() const { return http3_status_tracker_.isHttp3Broken(); }
+absl::optional<AlternateProtocolsCache::Origin> ConnectivityGrid::originFromHost() const {
+  if (host_->address()->type() != Network::Address::Type::Ip) {
+    ENVOY_LOG_MISC(error, "Address is not an IP address");
+    ASSERT(false);
+    return absl::nullopt;
+  }
+  uint32_t port = host_->address()->ip()->port();
+  // TODO(RyanTheOptimist): Figure out how scheme gets plumbed in here.
+  return AlternateProtocolsCache::Origin("https", host_->hostname(), port);
+}
+
+bool ConnectivityGrid::isHttp3Broken() const {
+  absl::optional<AlternateProtocolsCache::Origin> origin = originFromHost();
+  OptRef<AlternateProtocolsCache::Http3StatusTracker> http3_status_tracker =
+      alternate_protocols_->getOrCreateHttp3StatusTracker(*origin);
```

Comment:
Since we do this several times how about adding a getHttp3StatusTracker() helper method.